### PR TITLE
docs: cleanup documentation

### LINKED ISSUE #1893: ak.to_parquet has lost its docstring documentation
### Which documentation?

Python docstrings

### What needs to be documented?

On the new site: https://awkward-array.readthedocs.io/en/main/reference/generated/ak.to_parquet.html

On the old site: https://awkward-array.readthedocs.io/en/latest/_auto/ak.to_parquet.html

It's missing in the code itself; this is all that the docstring has:

https://github.com/scikit-hep/awkward/blob/d9dbab6a52896165568844a28d7eee4f05b2133a/src/awkward/operations/ak_to_parquet.py#L37-L68

It's probably the only one, because of all the work that the Parquet functions have been getting recently. But someone should do a quick scan through all of the operations to see if there are any more missing docstrings.

### PR DESCRIPTION
**Pull request by agoose77**
This PR 
- cleans up docstrings that were wonky, or missing fields. I didn't check them hugely for meaning; this was just a quick pass over the arguments lists.
- addresses quirks in user-guide 

I need to ping Martin to check some of the parquet/arrow parts.

Partially addresses #1916
Fixes #1893

----
:books: The documentation for this PR will be available at <https://awkward-array.readthedocs.io/en/agoose77-docs-touch-ups/> once Read the Docs has finished building :hammer:

### CODE REVIEW

**Comment by agoose77** [comment]
This only touches docs, so it doesn't run any of the build tests. There doesn't seem to be a sane way to specify conditional required tests. We could make all workflows run for every PR, and skip every step if the job is not required, but that will bloat all of our workflows

File: docs/user-guide/10-minutes-to-awkward-array.md [hunk 1]
```diff
@@ -85,71 +85,65 @@ In order to plot the taxi routes, we can use the waypoints given by the `path` f
...
-```{code-cell} ipython3
-x = ak.Array([1, 2]) # note the missing 3!
-y = ak.Array([
-    [4, 5, 6],
-    [7, 8, 9]
-])
+```{code-cell} python3
+x = ak.Array([1, 2])  # note the missing 3!
+y = ak.Array([[4, 5, 6], [7, 8, 9]])
 ak.broadcast_arrays(x, y)
```
Comment (jpivarski) [inline docs/user-guide/10-minutes-to-awkward-array.md]: I think you applied an automated code-formatter here, but the usefulness of this formatting was to explain broadcasting geometrically. It's not wrong to have them adhere to standard formatting here, but it's pedagogically _better_ for them to have the special formatting.

**Review by jpivarski — APPROVED** [review]
These are all improvements, with one exception, noted below. If it's hard to specify an exception to the auto-formatting (black?), it can stay.

File: docs/user-guide/10-minutes-to-awkward-array.md (hunk 1 above)
Comment (agoose77) [reply docs/user-guide/10-minutes-to-awkward-array.md]: Agreed, this is clearer. I've used the trailing-comma preservation behaviour to ensure that this is reverted and remains stable.

**Pull request merged**